src/components/KMultiselect/KMultiselect.vue

```diff
@@ -0,0 +1,911 @@
+          if (height > selectionsMaxHeight.value) {
+            const item = visibleSelectedItemsStaging.value.pop()
+            if (item) {
+              invisibleSelectedItemsStaging.value.push(item)
+            }
+          }
+          stagingKey.value++
+        }
+      }, 0)
+    }
+
+    // handles programmatic selections
+    const handleMultipleItemsSelect = (items: MultiselectItem[]) => {
+      selectedItems.value = []
+      visibleSelectedItemsStaging.value = []
+      items.forEach(itemToSelect => {
+        const selectedItem = unfilteredItems.value.filter(anItem => anItem.key === itemToSelect.key)[0]
+        selectedItem.selected = true
+        selectedItem.key = selectedItem?.key?.includes('-selected') ? selectedItem.key : `${selectedItem.key}-selected`
+        selectedItem.key += '-selected'
+        selectedItems.value.push(selectedItem)
+        visibleSelectedItemsStaging.value.push(selectedItem)
+      })
+
+      stageSelections()
+    }
+
+    // handle item select/deselect from dropdown
+    const handleItemSelect = (item: MultiselectItem) => {
+      const selectedItem = unfilteredItems.value.filter(anItem => anItem.key === item.key)[0]
+      // if clicked item is already selected
+      if (selectedItem.selected) {
+        selectedItems.value = selectedItems.value.filter(anItem => anItem.key !== item.key)
+        // remove item from visibility arrays
+        if (visibleSelectedItemsStaging.value.filter(anItem => anItem.key === item.key).length) {
+          visibleSelectedItemsStaging.value = visibleSelectedItemsStaging.value.filter(anItem => anItem.key !== item.key)
+        } else if (invisibleSelectedItemsStaging.value.filter(anItem => anItem.key === item.key).length) {
+          invisibleSelectedItemsStaging.value = invisibleSelectedItemsStaging.value.filter(anItem => anItem.key !== item.key)
+        }
+        // deselect item
+        selectedItem.selected = false
+        selectedItem.key = selectedItem?.key?.replace(/-selected/gi, '')
+
+        // if some items are hidden grab the first hidden one and add it into the visible array
+        if (invisibleSelectedItemsStaging.value.length) {
+          const item = invisibleSelectedItemsStaging.value.pop()
+          if (item) {
+            visibleSelectedItemsStaging.value.push(item)
+          }
+        }
+      } else { // newly selected item
+        selectedItem.selected = true
+        selectedItem.key = selectedItem?.key?.includes('-selected') ? selectedItem.key : `${selectedItem.key}-selected`
+        selectedItem.key += '-selected'
+        selectedItems.value.push(selectedItem)
+        visibleSelectedItemsStaging.value.push(selectedItem)
+      }
+
+      stageSelections()
+      const selectedVals = selectedItems.value.map(anItem => anItem.value)
+
+      emit('selected', selectedItems.value)
+      emit('update:modelValue', selectedVals)
+      // this 'input' event must be emitted for v-model binding to work properly
```

Comment:
```suggestion
      // this 'update:modelValue' event must be emitted for v-model binding to work properly
```

Reply:
Let's try removing `@input` events in the whole component, and just use `update:modelValue`